src/python/pants/base/build_root.py

```diff
@@ -36,10 +37,10 @@ def find_buildroot(self) -> str:
     return str(buildroot)
 
   def __init__(self) -> None:
```

Comment:
The sequence is `__new__` (which creates the instance and returns the class type), then `__init__` (which initializes the instance created by `__new__`: https://docs.python.org/3/reference/datamodel.html#basic-customization

Without looking at the doc or knowing that though, you might locally notice two things about `__init__`:
1. It's an instance method, not a classmethod or staticmethod.
2. In practice you'll find no implementations that return anything.


2.

Reply:
Ohh, got it! Thanks. It makes sense, seeing how `__init__` takes a `self`. Thanks a lot :)

Reply:
Yes, trippy indeed!

By the way, the reason we added `-> None` to this function—even though `__init__()` always has this return type and MyPy knows this by looking up the superclass `object()`—is so that we signal to MyPy to check the body of the function.

This is the main reason when writing tests that you want to use `-> None`.